crates/red_knot_python_semantic/src/types/display.rs

```diff
@@ -156,6 +156,7 @@ impl Display for DisplayRepresentation<'_> {
                 };
                 write!(f, "<wrapper-descriptor `{method}` of `{object}` objects>")
             }
+            Type::DataclassDecorator(_) => f.write_str("<function dataclasses.dataclass>"),
```

Comment:
This seems misleading, as this type does not represent the function `dataclasses.dataclass`, but rather some internal closure returned by it. The runtime says `<function dataclass.<locals>.wrap at 0x102892700>`.

I think the LSP folks would like us to avoid these angle-bracket names altogether, since they don't render nicely when syntax-highlighted as Python, but I'm not sure what alternative would work well here that would render nicely as Python. We may decide to just fallback to a generic callable signature type for this in the LSP?

For now maybe this would be less misleading:
```suggestion
            Type::DataclassDecorator(_) => f.write_str("<decorator produced by dataclasses.dataclass>"),
```

Reply:
> We may decide to just fallback to a generic callable signature type for this in the LSP?

That sounds reasonable, yes.

Using your suggestion for now.